utils/post.js

```diff
@@ -310,39 +310,54 @@ function modifyReactionsPost(post, isAnonimous = true) {
   return newPost;
 }
 
-async function filterFeeds(userId, feeds = [], feed_id = null) {
-  const newResult = [];
+const isExpiredPost = (item) => {
+  const now = moment().valueOf();
+  const dateExpired = moment(item?.expired_at).valueOf();
 
-  for (const item of feeds || []) {
-    const now = moment().valueOf();
-    const dateExpired = moment(item?.expired_at).valueOf();
+  return dateExpired < now;
+};
 
-    if (dateExpired < now) {
-      if (feed_id) {
-        deleteActivityFromUserFeed('topic', feed_id, item.id);
-      }
-      continue;
-    }
-    let newItem = {...item};
+const deleteExpiredPostFromFeed = (item, feed_id) => {
+  if (feed_id) {
+    deleteActivityFromUserFeed('topic', feed_id, item.id);
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `deleteExpiredPostFromFeed` is missing error handling. If the `deleteActivityFromUserFeed` function fails, it could lead to inconsistencies in the data. Consider wrapping this call in a try-catch block.
```diff
const deleteExpiredPostFromFeed = (item, feed_id) => {
  if (feed_id) {
    try {
      deleteActivityFromUserFeed('topic', feed_id, item.id);
    } catch (error) {
      console.error(`Failed to delete expired post from feed: ${error}`);
    }
  }
};
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->